library/std/src/panicking.rs

```diff
@@ -248,13 +248,7 @@ fn default_hook(info: &PanicInfo<'_>) {
     // The current implementation always returns `Some`.
     let location = info.location().unwrap();
 
-    let msg = match info.payload().downcast_ref::<&'static str>() {
-        Some(s) => *s,
-        None => match info.payload().downcast_ref::<String>() {
-            Some(s) => &s[..],
-            None => "Box<dyn Any>",
-        },
-    };
+    let msg = payload_as_str(info.payload());
```

Comment:
Can't this use `info.payload_as_str()`?

Reply:
I want to use the same fallback string (`"Box<dyn Any>"`) as in the other cases here, so I re-use the same helper function here. And that helper function doesn't use PanicHookInfo::payload_as_str(), because it is also used in places where there is no PanicHookInfo.

Reply:
Makes sense... looking at my own code doing these downcasts, it's mostly working in the return value from `catch_unwind`, where there's also no `PanicHookInfo`. Maybe we should have some more widely used helper method that works on all `Box<dyn Any + ...>` or `&dyn Any` or so.